To: vim_dev@googlegroups.com
Subject: Patch 7.3.459
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.459
Problem:    Win32: Warnings for type conversion.
Solution:   Add type casts. (Mike Williams)
Files:	    src/misc2.c, src/os_win32.c


*** ../vim-7.3.458/src/misc2.c	2012-02-22 18:12:29.000000000 +0100
--- src/misc2.c	2012-02-29 13:44:17.000000000 +0100
***************
*** 2074,2080 ****
  	    n = gap->ga_growsize;
  	new_len = gap->ga_itemsize * (gap->ga_len + n);
  	pp = (gap->ga_data == NULL)
! 			? alloc(new_len) : vim_realloc(gap->ga_data, new_len);
  	if (pp == NULL)
  	    return FAIL;
  	old_len = gap->ga_itemsize * gap->ga_maxlen;
--- 2074,2080 ----
  	    n = gap->ga_growsize;
  	new_len = gap->ga_itemsize * (gap->ga_len + n);
  	pp = (gap->ga_data == NULL)
! 	      ? alloc((unsigned)new_len) : vim_realloc(gap->ga_data, new_len);
  	if (pp == NULL)
  	    return FAIL;
  	old_len = gap->ga_itemsize * gap->ga_maxlen;
*** ../vim-7.3.458/src/os_win32.c	2012-02-22 15:34:05.000000000 +0100
--- src/os_win32.c	2012-02-29 13:43:39.000000000 +0100
***************
*** 264,270 ****
      static void
  unescape_shellxquote(char_u *p, char_u *escaped)
  {
!     int	    l = STRLEN(p);
      int	    n;
  
      while (*p != NUL)
--- 264,270 ----
      static void
  unescape_shellxquote(char_u *p, char_u *escaped)
  {
!     int	    l = (int)STRLEN(p);
      int	    n;
  
      while (*p != NUL)
*** ../vim-7.3.458/src/version.c	2012-02-29 13:51:32.000000000 +0100
--- src/version.c	2012-02-29 13:58:08.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     459,
  /**/

-- 
"A clear conscience is usually the sign of a bad memory."
                             -- Steven Wright

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
